packages/expo-modules-autolinking/scripts/ios/project_integrator.rb

```diff
@@ -128,5 +132,124 @@ def self.set_autolinking_configuration(project)
       end
     end
 
-  end # module ExpoAutolinkingExtension
+    # Makes sure that the build script configuring the project is installed,
+    # is up-to-date and is placed before the "Compile Sources" phase.
+    def self.integrate_build_script(autolinking_manager, project, target, native_target)
+      build_phases = native_target.build_phases
+      modules_provider_path = autolinking_manager.modules_provider_path(target)
+
+      # Look for our own build script phase
+      xcode_build_script = native_target.shell_script_build_phases.find { |script|
+        script.name == Constants::CONFIGURE_PROJECT_BUILD_SCRIPT_NAME
+      }
+
+      if xcode_build_script.nil?
+        # Inform the user that we added a build script.
+        puts "[Expo] ".blue << "Installing the build script for target " << native_target.name.green
+
+        # Create a new build script in the target, it's added as the last phase
+        xcode_build_script = native_target.new_shell_script_build_phase(Constants::CONFIGURE_PROJECT_BUILD_SCRIPT_NAME)
+      end
+
+      # Make sure it is before the "Compile Sources" build phase
+      xcode_build_script_index = build_phases.find_index(xcode_build_script)
+      compile_sources_index = build_phases.find_index { |phase|
+        phase.is_a?(Xcodeproj::Project::PBXSourcesBuildPhase)
+      }
+
+      # Insert our script before the "Compile Sources" phase when necessary
+      unless compile_sources_index.nil? || xcode_build_script_index < compile_sources_index
+        build_phases.insert(
+          compile_sources_index,
+          build_phases.delete_at(xcode_build_script_index)
+        )
+      end
```

Comment:
```suggestion
      xcode_build_script_index = build_phases.find_index(xcode_build_script)
      compile_sources_index = build_phases.find_index { |phase|
        phase.is_a?(Xcodeproj::Project::PBXSourcesBuildPhase)
      }
      raise 'Unable to find xcode_build_script_index' if xcode_build_script_index.nil?
      raise 'Unable to find compile_sources_index' if compile_sources_index.nil?
      
      # Insert our script before the "Compile Sources" phase when necessary
      unless xcode_build_script_index < compile_sources_index
        build_phases.insert(
          compile_sources_index,
          build_phases.delete_at(xcode_build_script_index)
        )
      end
```

maybe defensively check the `xcode_build_script_index` is not nil as well.
what will happen if `compile_sources_index` is nil? does the provide generate after source compiling? so i'm proposing raising an exception when `compile_sources_index` is nil.

Reply:
I'm not sure if we should be raising errors, especially in this script. I would rather prefer just logging the issue instead of making the `pod install` to fail (another reason to blame expo for something 😅). Take into account that installing this build script is not really crucial right now as it's necessary only for users having their own local modules.

`xcode_build_script_index` cannot be nil as it was just added to the array by `native_target.new_shell_script_build_phase` if it wasn't there already.

If `compile_sources_index` is nil then I believe Xcode won't be able to build the project 😉

Reply:
> I'm not sure if we should be raising errors, especially in this script. 

that's a good point. nice work for logging rather than throw errors.